### ISSUE TITLE
Training: add validation set to check for overtraining

### ISSUE THREAD

**Issue by stefanie-marti** [stefanie-marti/GammaSpaghetti#2]
Training progress can only be monitored via the performance on the training set. However, this does not allow to detect a possible overtraining. So a small part of the training set should be split and used as a validation set, or a validation set should be loaded.
